zigzag/include/zigzag.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+#include <cairo.h>
+#include <wayland-server-core.h>
+#include <wlr/render/wlr_texture.h>
+#include <wlr/util/box.h>
+
+struct zigzag_node;
+
+struct zigzag_layout_impl {
+  void (*on_damage)(struct zigzag_node *node);
+};
+
+struct zigzag_layout {
+  int output_width;
+  int output_height;
+
+  struct wl_list nodes;  // zigzag_node::link
+
+  void *user_data;
+
+  const struct zigzag_layout_impl *implementation;
+};
+
+struct zigzag_layout *zigzag_layout_create(
+    const struct zigzag_layout_impl *implementation, int output_width,
+    int output_height, void *user_data);
+
+void zigzag_layout_destroy(struct zigzag_layout *self);
+
+struct zigzag_node_impl {
+  void (*on_click)(struct zigzag_node *self, double x, double y);
+  void (*set_frame)(
+      struct zigzag_node *self, int output_width, int output_height);
+  void (*render)(struct zigzag_node *self, cairo_t *cr);
+};
+
+struct zigzag_node {
+  struct zigzag_layout *layout;
+
+  struct wlr_box *frame;
+  struct wlr_texture *texture;
+
+  void *user_data;
+
+  struct wl_list link;  // for zigzag_layout :: nodes
+  struct wl_list children;
```

Comment:
```suggestion
  struct wl_list node_list;
``` 
We preffer `*_list` for `wl_list`, and `*s` for `wl_array`.